solutions/cses-1623.mdx

```diff
@@ -0,0 +1,150 @@
+of the smallest difference between the sum of the weights of the apples that
+are part of the current combination and the sum of the weights that are part of
+the complement combination. In order to go through all of the cominations, we
+use backtracking to go through both adding the current index and not adding it
+for each index in the input array.
+
+### Implementation
+
+**Time complexity:** O(2^{N}) because we are doing one update to the result
+for each of the 2^{N} combinations
+
+<LanguageSection>
+<JavaSection>
+```java
+import java.util.*;
+import java.io.*;
+
+class Sol {
+    private static ArrayList<Integer> currentSubset = new ArrayList<Integer>();
+    // initialize result to max value because we want to get the minimum result
+    private static double result = Integer.MAX_VALUE;
+    private static long total = 0;
+
+    public static void main(String[] args) {
+        Kattio kattio = new Kattio();
+        int N = kattio.nextInt();
+        int[] weights = new int[N];
+
+        for (int i = 0; i < N; i++) {
+            weights[i] = kattio.nextInt();
+	    total += weights[i]; // calculate the total weight of all apples
+        }
+
+        completeSearch(weights, 0, N, 0);
+	System.out.println((int)(result * 2));
+    }
+
+    public static void completeSearch(int[] weights, int i, int N,
+    	   long currentSum) {
+        if (i == N) {
+	    result = Math.min(result, Math.abs((total - currentSum)
+	    	   - currentSum) / 2.0);
+        } else {
+	    // option 1: not adding current index
+            completeSearch(weights, i + 1, N, currentSum);
+
+	    // option 2: adding current index
+            currentSubset.add(i);
+	    currentSum += weights[i];
+            completeSearch(weights, i + 1, N, currentSum);
+	    // undo option 2
+	    currentSum -= currentSubset.get(currentSubset.size() - 1);
+            currentSubset.remove(currentSubset.size() - 1);
+        }
+    }
+    CodeSnip{Kattio}
+}
+```
+</JavaSection>
+
+<CPPSection>
+```cpp
+#include<bits/stdc++.h>
+using namespace std;
+
+vector<int> currentSubset;
+// initialize result to max value because we want to get the minimum result
+double result = INT_MAX;
+long int total = 0;
+
+void completeSearch(vector<int> &weights, int i, int N, long int currentSum) {
+    if (i == N) {
+        result = min(result, abs((total - currentSum) - currentSum) / 2.0);
+    } else {
+        // option 1: not adding current index
+        completeSearch(weights, i + 1, N, currentSum);
+
+	// option 2: adding current index
+        currentSubset.push_back(i);
+        currentSum += weights[i];
+        completeSearch(weights, i + 1, N, currentSum);
+	// undo option 2
+        currentSum -= currentSubset[currentSubset.size() - 1];
+        currentSubset.pop_back();
+    }
+}
+
+int main() {
+    int N; cin >> N;
+    vector<int> weights(N);
+
+    for (int i = 0; i < N; i++) {
+        cin >> weights[i];
+	total += weights[i]; // calculate the total weight of all apples
```

Comment:
inconsistent indentation (also 2 spaces after the line of code for an inline comment please)